File: CheckPower.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
bool checkPower(int x, int y) {
    if (x == 0) return (y == 0);  // 0^k = 0 với k > 0, nhưng không có 0^0
    if (x == -1) return (y == 1 || y == -1);  // -1^k = 1 với mọi k chẵn và =-1 với mọi k lẻ
    if (y == 1) return true;  // x^0 = 1 với mọi x
    if (x > 0 && y < 0) return false;  // Không thể có lũy thừa dương bằng số âm
    if (x < 0) {
        int sum = 1;
        while ((y < 0 && abs(sum) < abs(y)) || (y > 0 && sum < y)){
            sum *= x;
            if (sum == y) return true;  // Kiểm tra nếu sum đã bằng y
        }
        return false;  // Nếu không tìm thấy lũy thừa phù hợp, trả về false
    }
    // Nếu x là số dương
    int sum = 1;
    while (sum < y) {
        sum *= x;
    }
    return (sum == y);
}

int main() {
    int x, y;
    printf("Hãy nhập hai số x và y bất kì cách nhau khoảng trắng: ");
    scanf("%d %d", &x, &y);
    if (checkPower(x, y)) {
        printf("True\n");
    } else {
        printf("False\n");
    }

    return 0;
}

File: CheckPower.py
import math

def checkPower(x,y):
    if x == 0: return (y == 0)
    if y == 1: return True
    if x == -1: return (y == 1 or y == -1)
    if x > 0 and y < 0: return False
    if x < 0:
        sum = 1
        while (y < 0 and abs(sum) < abs(y)) or (y > 0 and sum < y):
            sum *= x
            if sum == y:
                return True
        return False
    power = 1
    while power < y:
        power *= x
    return (power == y)

x,y = map(int, input("Hãy nhập 2 số x, y bất kì: ").split())
if checkPower(x,y):
    print("True")
else: print("False")